test/functional/test_framework/test_framework.py

```diff
@@ -150,31 +149,28 @@ def main(self):
         else:
             self.log.warning("Not cleaning up dir %s" % self.options.tmpdir)
             if os.getenv("PYTHON_DEBUG", ""):
-                # Dump the end of the debug logs, to aid in debugging rare
-                # travis failures.
-                import glob
-                filenames = [self.options.tmpdir + "/test_framework.log"]
-                filenames += glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
-                MAX_LINES_TO_PRINT = 1000
-                for fn in filenames:
-                    try:
-                        with open(fn, 'r') as f:
-                            print("From", fn, ":")
-                            print("".join(deque(f, MAX_LINES_TO_PRINT)))
-                    except OSError:
-                        print("Opening file %s failed." % fn)
-                        traceback.print_exc()
+                logging.shutdown()  # Flush handlers
+                lines = 4000
+                print()
+                print('Combine the logs and print the last {} lines ...'.format(lines))
```

Comment:
nit: I have a preference for:

`print('\nCombine the logs and print the last {} lines ...\n'.format(lines))`

rather than the empty `print()` function calls.